### PR TITLE
Add locale picker to header

### PR DESCRIPTION
**Pull request by nscuro**
### Description

Adds a locale picker to the header of the page, next to the profile icon.

This is to make it easier to switch languages, without having to fiddle with browser settings.

### Addressed Issue

N/A

### Additional Details

Current version (don't ask me why the GIF quality is worse, no idea):

Kapture 2024-04-23 at 19 54 08

Original version:

Kapture 2024-04-22 at 22 44 58

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#pull-requests)
- ~This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

**Comment by nscuro** [comment]
@fnxpt Fancy giving this a review?

Besides the code itself, please have a look at the new translations. For German, the auto-translate suggested "polieren" for "Polish" (as in polishing, the verb lol).

File: src/i18n/locales/pt.json [hunk 1]
```diff
@@ -293,6 +293,20 @@
     "sha_384": "SHA-384",
     "sha_512": "SHA-512"
   },
+  "language": {
+    "de": "Alemão",
```
Comment (fnxpt) [inline src/i18n/locales/pt.json:297]: The text is all correct, I would just put the first char as uppercase on all of them

File: src/i18n/locales/pt.json [hunk 2]
```diff
@@ -500,6 +516,7 @@
     "inventory_with_vulnerabilities": "Inventário com vulnerabilidades",
     "justification": "Justificação",
     "justification_tooltip": "A justificativa pela qual o estado da análise de impacto foi declarado como \"Não afetado\"",
+    "language": "Linguagem",
```
Comment (fnxpt) [inline src/i18n/locales/pt.json:519]: Lingua

File: src/i18n/locales/pt-BR.json [hunk 3]
```diff
@@ -500,6 +516,7 @@
     "inventory_with_vulnerabilities": "Inventário com vulnerabilidades",
     "justification": "Justificação",
     "justification_tooltip": "A justificativa pela qual o estado da análise de impacto foi declarado como \"Não afetado\"",
+    "language": "Linguagem",
```
Comment (fnxpt) [inline src/i18n/locales/pt-BR.json:519]: Lingua

File: src/i18n/locales/pt-BR.json [hunk 4]
```diff
@@ -293,6 +293,20 @@
     "sha_384": "SHA-384",
     "sha_512": "SHA-512"
   },
+  "language": {
+    "de": "Alemão",
```
Comment (fnxpt) [inline src/i18n/locales/pt-BR.json:297]: The text is all correct, I would just put the first char as uppercase on all of them

**Review by fnxpt — COMMENTED** [review]
As an additional comment, I would add flags instead of text, usually on these types of selectors its more common to see a flag

File: src/i18n/locales/es.json [hunk 5]
```diff
@@ -293,6 +293,20 @@
     "sha_384": "SHA-384",
     "sha_512": "SHA-512"
   },
+  "language": {
+    "de": "Alemán",
+    "en": "Inglés",
+    "es": "Español",
+    "fr": "Francés",
+    "hi": "hindi",
```
Comment (fnxpt) [inline src/i18n/locales/es.json:301]: The text is all correct, I would just put the first char as uppercase on all of them

**Comment by fnxpt** [comment]
> Good point. I briefly looked into that but couldn't be bothered to add yet another NPM package just for flag icons. I'll have another look, I agree that country codes might be sub-optimal.
We can also add the images on the resources and just do {locale}.png, no need for an extra library

**Comment by nscuro** [comment]
@fnxpt Any good source for flag icons you can recommend?

**Comment by nscuro** [comment]
Languages are now displayed as country flag emojis instead of language codes, as per @fnxpt's suggestion: 

image

Got feedback from @msymons that the picker should probably be located closer to the user profile. I'll see if I can add it to the dropdown that opens when clicking the user icon.

File: src/i18n/locales/pt.json (hunk 1 above)
Comment (nscuro) [reply src/i18n/locales/pt.json:297]: Done.

File: src/i18n/locales/pt.json (hunk 2 above)
Comment (nscuro) [reply src/i18n/locales/pt.json]: Thanks! Done.

File: src/i18n/locales/pt-BR.json (hunk 3 above)
Comment (nscuro) [reply src/i18n/locales/pt-BR.json]: Done.

File: src/i18n/locales/pt-BR.json (hunk 4 above)
Comment (nscuro) [reply src/i18n/locales/pt-BR.json:297]: Done.

File: src/i18n/locales/es.json (hunk 5 above)
Comment (nscuro) [reply src/i18n/locales/es.json]: Done.

**Comment by nscuro** [comment]
@fnxpt Suggestions applied. Updated the GIF in the PR description to show the current look&feel of the language picker. Can you have another look please?

**Review by fnxpt — APPROVED** [review]

**Comment by fnxpt** [comment]
lgtm

**Pull request merged**